server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -283,6 +281,43 @@ public boolean setExpiration(String name, Instant expiration) {
     return Optional.empty();
   }
 
+  /**
+   * @param names of secrets series to look up secrets by.
+   * @return Secrets matching input parameters.
+   */
+  public List<SecretSeriesAndContent> getSecretsByName(List<String> names) {
+    checkArgument(!names.isEmpty());
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    List<SecretSeries> multipleseries = secretSeriesDAO.getSecretsSeriesByName(names);
+
+
+    Iterator siter = multipleseries.iterator();
+
+    List<SecretSeriesAndContent> ret = new ArrayList<SecretSeriesAndContent>();
+
+    while (siter.hasNext()) {
```

Comment:
Rather than using iterators (here and in `BatchSecretDeliveryResource`), I generally prefer a foreach loop:
```
for (SecretSeries series : multipleSeries) {
  ...
}
```
It's a little easier to read and avoids possible mistakes from forgetting to check `hasNext()`/not calling `next()`.